### PR TITLE
Spawner Profile page

### PR DESCRIPTION
**Pull request by kildre**
Created a page where the user can select a server type they would like to use for their application

## What does this implement/fix?
This removes the application server type selection from the form and puts it on its own page.
_Put a `x` in the boxes that apply_

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds a feature)
- [ ] Breaking change (fix or feature that would cause existing features not to work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] Other (please describe):

## Testing

- [x] Did you test the pull request locally?
- [x] Did you add new tests?

#### Text styling

- [x] The content is written with [plain language](https://www.plainlanguage.gov/guidelines/) (where relevant).
- [x] If there are headers, they use the proper header tags (with only one level-one header: `H1` or `#` in markdown).
- [x] All links describe where they link to (for example, check the [Nebari website](https://nebari.dev/)).
- [x] This content adheres to the Nebari style guides.

#### Non-text content

- [ ] All content is represented as text (for example, images need alt text, and videos need captions or descriptive transcripts).
- [ ] If there are emojis, there are not more than three in a row.
- [ ] Don't use [flashing GIFs or videos](https://www.w3.org/TR/UNDERSTANDING-WCAG20/seizure-does-not-violate.html).
- [x] If the content were to be read as plain text, it still makes sense, and no information is missing.

## Any other comments?
server-types_loading
server-types_servers

### CODE REVIEW

File: ui/src/components/radio-button/radio-button.tsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  isTile,
+  onChange,
+  onClick,
+  label,
+  subtext,
+  value,
+  checked,
+}) => {
+  // Determine the class for styling based on `isTile` prop
+  const className = isTile ? 'radio-tile' : 'radio-button';
```
Comment (jbouder) [inline ui/src/components/radio-button/radio-button.tsx:55]: I'm not seeing this `radio-tile` button anywhere in the CSS. I'd suggest removing the tile prop and references as we will be replacing this with MUI soon

File: ui/src/index.css [hunk 2]
```diff
@@ -2,6 +2,92 @@
...
+
+/* Active state */
+a:active {
+  color: #03396c; /* An even darker shade for active/clicked state */
+}
+
+/* Visited state */
+a:visited {
+  color: #4b2e83; /* A purple color for visited links */
+}
```
Comment (jbouder) [inline ui/src/index.css:34]: This stying is going to affect the entire app, unless these changes should be across the board, you need to scope these or rename

File: ui/src/index.css [hunk 3]
```diff
@@ -2,6 +2,92 @@
...
+
+.button-container {
+  display: flex;
+  justify-content: space-between;
+  margin: 2rem 0;
+  padding: 2rem 0;
+}
+.button-container button {
+}
```
Comment (jbouder) [inline ui/src/index.css:85]: Do should be removed

File: ui/src/index.css [hunk 4]
```diff
@@ -2,6 +2,92 @@
...
+
+.button-group {
+  display: flex;
+}
```
Comment (jbouder) [inline ui/src/index.css:89]: this class is defined further down the file, should be ok to remove it

File: ui/src/pages/server-types/server-types.tsx [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+    name: 'Large Instance',
+    subtext: 'Stable Environment with 6 CPU / 32GB RAM',
+  },
+  {
+    id: '7',
+    name: 'Large Instance',
+    subtext: 'Stable Environment with 8 CPU / 40GB RAM',
+  },
+  {
+    id: '8',
```
Comment (jbouder) [inline ui/src/pages/server-types/server-types.tsx:48]: This component however should be loading the server types from the `/spawner-profiles/` endpoint.

File: ui/src/pages/server-types/server-types.tsx [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+    // Submit the form data, for example, to an API
+    console.log('Selected server type:', selectedServerType);
+  };
+
+  return (
+    <form onSubmit={handleSubmit} className="container mx-auto px-4">
+      <div className="container mx-auto px-4">
+        <div className="grid grid-cols-12 gap-4">
+          <div className="col-start-4 col-span-6 bg-gray-200 p-4">
+            <h1 className="mb-0">Hardware Environment</h1>
```
Comment (jbouder) [inline ui/src/pages/server-types/server-types.tsx:73]: This should be Server Type...and maybe update the paragraph wording below

**Review by jbouder — CHANGES_REQUESTED** [review]
See comments for some changes. Also make sure to do another buildCopy once all updates are made

File: ui/src/pages/server-types/server-types.tsx (hunk 5 above)
Comment (jbouder) [reply ui/src/pages/server-types/server-types.tsx:48]: These keys should be updated to match the endpoint as well. Here is the current mocked data: https://github.com/nebari-dev/jhub-apps/blob/f3ce17cf3e092476e94e33e2de509d6184ec4593/ui/src/data/api.ts#L17.

Please add descriptions to the existing items. And might also be good to add some additional items like you have to the dummy data to illustrate a larger list of server types

File: ui/src/components/radio-button/radio-button.tsx (hunk 1 above)
Comment (kildre) [reply ui/src/components/radio-button/radio-button.tsx]: Scrapped this code from comet library, removed

**Comment by dharhas** [comment]
Launching JupyterLab or VSCode will also hit the spawn page. Are those going to be the same as this one or different?

Reason I'm asking is "create app" doesn't fit in those cases.

File: ui/src/pages/server-types/server-types.tsx [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+    axios
+      .get('/spawner-profiles/')
+      .then((response) => {
+        console.log('Full response:', response);
+        return setServerTypes(response.data);
+      })
+      .catch((error) => {
+        console.error('Error fetching data:', error);
+      });
+  }, []);
```
Comment (jbouder) [inline ui/src/pages/server-types/server-types.tsx:20]: You should be using react-query for making API calls, should be no need for a useEffect. See here for an example: https://github.com/nebari-dev/jhub-apps/blob/f3ce17cf3e092476e94e33e2de509d6184ec4593/ui/src/pages/home/apps-grid/apps-grid.tsx#L25

File: ui/src/pages/server-types/server-types.tsx [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+
+  useEffect(() => {
+    console.log('serverTypes', serverTypes);
+  }, [serverTypes]);
```
Comment (jbouder) [inline ui/src/pages/server-types/server-types.tsx:24]: Remove this, we shouldn't be logging to the console outside of development

**Review by jbouder — CHANGES_REQUESTED** [review]
In addition to the comments, please also review unit tests. Looks like 1 might be failing.

**Review by jbouder — APPROVED** [review]
Great work!

**Review by aktech — APPROVED** [review]
Preview looks great! 🎉

**Pull request merged**